README.md

```diff
@@ -1,133 +1,114 @@
-new tensor with the shape `(1, 256)`, by concatenating the tensors and either pad the remaining space or trimming it.
-The new tensor is the reference data for the models which expect a reference data with shape of `(1, 256)` and transaction details tensor with the shape of `(1, 30)`. Once the inferencing of the models are done, the function uses `numpy` to aggregate the results and save them to a tensor
-with shape `(1, 2)` that contains the probability for the transaction to be a fraud.
-
-![Gears<->AI execution](./flow4.png "Gears<->AI execution")
+This demo showcases the advantage of **data locality** during transaction scoring, by using [RedisAI](https://oss.redislabs.com/redisai/) module.
+It simulates a fraud-detection app which is based on ML model prediction in real-time. When a new transaction is executed, it triggers a flow whose output is the probability that this transaction is fraudulent. The prediction is based on both the given transaction and some reference data stored in Redis, while *the entire flow* is done within Redis as well.   
 
 ## Running the Demo
-To run the demo:
+To run the demo app and load the data, run the following:
 ```
-$ git clone https://github.com/RedisAI/FraudDetectionDemo.git
-$ cd FraudDetectionDemo
 # If you don't have it already, install https://git-lfs.github.com/ (On OSX: brew install git-lfs)
 $ git lfs install && git lfs fetch && git lfs checkout
+
+# Clone the repository
+$ git clone https://github.com/RedisAI/FraudDetectionDemo.git
+$ cd FraudDetectionDemo
+
+# Launch the demo with docker-compose (may require sudo in linux)
 $ docker-compose up
 ```
 If something went wrong, e.g. you skipped installing git-lfs, you need to force docker-compose to rebuild the containers
 ```
 $ docker-compose up --force-recreate --build
 ```
 
-The `app` container will [load](https://github.com/RedisAI/FraudDetectionDemo/blob/master/app/app.py)
-- The [two tensorflow models](https://github.com/RedisAI/FraudDetectionDemo/tree/master/app/models)
-- The [Torch script](https://github.com/RedisAI/FraudDetectionDemo/blob/master/app/script.torch)
-- The RedisGears functions
+## Architecture
+### RedisAI
+Redis instance is launched with RedisAI module loaded (by running `redislabs/redisai:latest` docker). RedisAI allows you to store and execute AI/ML pre-trained models in Redis.
+
+### [Data Loader](https://github.com/RedisAI/FraudDetectionDemo/blob/master/dataloader/load.py)
+Historical data of credit card transactions is loaded into Redis from [`data/creditcard.csv` file](https://media.githubusercontent.com/media/RedisAI/FraudDetectionDemo/master/dataloader/data/creditcard.csv), using redis-py (Redis' official python client). Every transaction is [stored in Redis](https://github.com/RedisAI/FraudDetectionDemo/blob/master/dataloader/load.py#L31) as a hash with the following structure:
```

Comment:
What kind of historical data is this? I presume some feature data?

Reply:
I guess so... The 29 numeric values are in the range of `(0, 1)` and have no clear meaning... how would you call it?

Reply:
model features! 

I imagine that These numbers are the result of some feature engineering process that takes raw features as inputs!

Reply:
I rephrased to "Historical feature-data". ok?

Reply:
or maybe "Historical data of credit card transactions which has been processed into features is loaded..."